lmdeploy/serve/openai/api_server.py

```diff
@@ -434,10 +447,24 @@ async def chat_completions_v1(request: ChatCompletionRequest,
         stop_words=request.stop,
         skip_special_tokens=request.skip_special_tokens)
 
+    tools = None
+    if request.tools and request.tool_choice != 'none':
+        if request.stream is True:
+            logger.warning('Set stream to False for tools')
+            request.stream = False
```

Comment:
why change it to `False`?

Reply:
I suppose it only supports stream=False. In streaming mode, how can client call the function? And it would be hard to extract content inside <plugin> too.

Reply:
Can the client call the function when the finish_reason is `tool_calls`,